dask/array/reductions.py

```diff
@@ -656,14 +685,28 @@ def cumreduction(func, binop, ident, x, axis, dtype=None, out=None):
     return handle_out(out, result)
 
 
+def _cumsum_merge(a, b):
+    if isinstance(a, np.ma.masked_array) or isinstance(b, np.ma.masked_array):
+        values = np.ma.getdata(a) + np.ma.getdata(b)
+        return np.ma.masked_array(values, mask=np.ma.getmaskarray(b))
```

Comment:
No, for cumulative operations the mask stays fixed throughout and the operation only occurs on the data. In this case `b` is whatever chunk already existed at that chunk-location, and `a` is the results from cumsum/cumprod all previous chunks along the axis.